subprojects/core/src/main/java/org/gradle/initialization/ScriptEvaluatingSettingsProcessor.java

```diff
@@ -65,7 +70,12 @@ public SettingsState process(
         SettingsInternal settings = state.getSettings();
         gradle.getBuildListenerBroadcaster().beforeSettings(settings);
         settings.getCaches().finalizeConfiguration(gradle);
-        applySettingsScript(settingsScript, settings);
+        try{
+            applySettingsScript(settingsScript, settings);
+        } catch (LocationAwareException e) {
```

Comment:
❓ Does this also handle runtime exceptions, or only script compilation exception atm?